internal/restic/backend.go

```diff
@@ -6,6 +6,11 @@ import (
 )
 
 // Backend is used to store and access data.
+//
+// Backend operations that return an error will be retried when a Backend is
+// wrapped in a RetryBackend. To prevent that from happening, the operations
+// should return a backoff.PermanentError. Errors from the context package
+// are always considered permanent.
```

Comment:
I don't think that's the case:

According to https://github.com/cenkalti/backoff/blob/18fe4ce5a8550e0d0919b680ad3c080a5455bddf/retry.go#L56
```
if permanent, ok := err.(*PermanentError); ok {
```

As `PermanentError` is a named struct from the `backoff` package, it won't consider a context cancelled error or similar as permanent. (If the used context itself is cancelled then the retries will be aborted, however that is independent of the context error itself).

Reply:
You're right. I changed the comment.